### PR TITLE
TEST: QA-10885 Automation test cover Add ability to treat null values as 0 when using MVF (R4 - UI SDK)

### PR DESCRIPTION
**Pull request by KhoaPham1010**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [ ] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Successful `extended test - examples`
- [ ] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
- gdc-dashboards:

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7381/ : FAILURE in 1m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3400/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7382/ : FAILURE in 2m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3401/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7383/ : FAILURE in 1m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3402/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7384/ : FAILURE in 1m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3403/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7385/ : FAILURE in 2m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3404/ : SUCCESS in 5s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7386/ : FAILURE in 1m 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3405/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7387/ : FAILURE in 11m 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3406/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7388/ : SUCCESS in 6m 46s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3407/ : SUCCESS in 22s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7389/ : SUCCESS in 3m 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3408/ : SUCCESS in 3s

**Comment by KhoaPham1010** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1607.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3409/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3218/ : FAILURE in 17m 38s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7391/ : SUCCESS in 2m 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3413/ : SUCCESS in 6s

**Comment by KhoaPham1010** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1607.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3414/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3221/ : SUCCESS in 17m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7394/ : SUCCESS in 4m 42s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3417/ : SUCCESS in 6s

**Comment by KhoaPham1010** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1607.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3419/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3222/ : SUCCESS in 22m 25s

File: examples/src/routes/VisualizationByIdentifier.jsx [hunk 1]
```diff
@@ -144,6 +146,12 @@ export const VisualizationByIdentifier = () => (
             for={VisualizationMeasureValueFilterByIdentifierExample}
             source={VisualizationMeasureValueFilterExampleSRC}
         />
+
+        <h2 id="measure-value-filter-treat-null-as-zero">Measure Value Filter Treat Null As Zero</h2>
```
Comment (xMort) [inline examples/src/routes/VisualizationByIdentifier.jsx:150]: Maybe change it to "Measure Value Filter that treats measure null values as zero"

File: examples/test/MeasureValueFilterComponent_test.js [hunk 2]
```diff
@@ -71,3 +74,19 @@ test("should filter dropdown with custom button", async t => {
...
+test("should filter the data with treat null value as zero", async t => {
+    const pivotTableSelector = Selector(".s-measure-value-filter-example-5");
+    await t.click(comparisonFilterDropdownSelector.nth(4));
+    await t.click(comparisonFilterOperatorDropdownSelector);
+    await t.click(lessThanIconSelector);
+    await t.typeText(comparisonInputSelector, "10");
+    await t.click(applySelector);
+
+    await waitForPivotTableStopLoading(t, pivotTableSelector);
+    // check applied filter
```
Comment (xMort) [inline examples/test/MeasureValueFilterComponent_test.js:87]: Either remove this comment or remove the blank line after it (or put it above it) as the comment belongs to the next block.

File: examples/test/MeasureValueFilter_test.js [hunk 3]
```diff
@@ -164,3 +164,15 @@ test("should render data when chart is configured", async t => {
         .expect(chartValues.nth(3).textContent)
         .eql("-24,823,451");
 });
+
+fixture("Measure Value Filter Treat Null As Zero").beforeEach(
```
Comment (xMort) [inline examples/test/MeasureValueFilter_test.js:168]: Optional: "Measure Value Filter that treats null as zero"

File: examples/test/MeasureValueFilter_test.js [hunk 4]
```diff
@@ -164,3 +164,15 @@ test("should render data when chart is configured", async t => {
...
+    loginUserAndNavigate(`${config.url}/hidden/measure-value-filter-with-treat-null-as-zero`),
+);
+
+test("should filter the data with measure value filter treat null as zero", async t => {
```
Comment (xMort) [inline examples/test/MeasureValueFilter_test.js:172]: Optional: "should filter the data with measure value filter that treats measure null values as zero"

File: examples/test/VisualizationByIdentifier_test.js [hunk 5]
```diff
@@ -110,3 +114,14 @@ test("Measure value filter should render", async t => {
         .expect(chartValues.nth(1).textContent)
         .eql("57,264,327.95");
 });
+
+test("Measure value filter with treat null value as zero should render", async t => {
+    const pivotTableSelector = Selector("#measure-value-filter-treat-null-as-zero ~ div .s-pivot-table");
+
+    await waitForPivotTableStopLoading(t, pivotTableSelector);
+    // check applied filter
```
Comment (xMort) [inline examples/test/VisualizationByIdentifier_test.js:122]: Please read my comment for similar issue above (blank line below the comment line).

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7399/ : FAILURE in 6m 57s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3423/ : SUCCESS in 3s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7402/ : SUCCESS in 5m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3426/ : SUCCESS in 3s

**Comment by KhoaPham1010** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1607.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3427/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3223/ : SUCCESS in 17m 43s

**Comment by xMort** [comment]
LGTM

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3866/ : SUCCESS in 13m 48s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/25/ : SUCCESS in 17s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7409/ : SUCCESS in 5m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1800/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2646/ : FAILURE in 14m 57s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3871/ : SUCCESS in 10m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/29/ : SUCCESS in 19s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7415/ : FAILURE in 7m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1804/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2651/ : SUCCESS in 21m 33s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3875/ : SUCCESS in 17m 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-fossa-scanning-zuul-el7/33/ : SUCCESS in 22s (non-voting)
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7421/ : SUCCESS in 10m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1808/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2655/ : SUCCESS in 22m 16s

**Pull request merged**